components/TaskCard.vue

```diff
@@ -63,30 +76,71 @@ export default {
       default: null,
     },
   },
-  computed: {
-    ...mapGetters({ requestInProcess: 'todos/getCompleteRequest' }),
-  },
-
   data: () => ({
     task: null,
+    showEditIcon: true,
+    titleInputError: false,
+    titleErrorMsg: '',
     loading: false,
   }),
+  computed: {
+    ...mapGetters({ requestInProcess: 'todos/getCompleteRequest' }),
+  },
 
   created() {
-    this.task = _.clone(this.cardData);
+    this.task = _.clone(this.cardData); // this.task will be edited . to avoid issues thrown by vuex instead to directly assigning the ref to this.task, cloning it.
   },
   methods: {
     markDone() {
-      this.loading = true;
-      this.$store.dispatch('todos/changeTaskState', this.task).then(() => {
-        if (!this.requestInProcess) {
-          this.loading = false;
-          this.task = _.clone(this.cardData);
-        }
-      });
+      if (this.showEditIcon) {
+        this.loading = true;
+        this.$store.dispatch('todos/changeTaskState', this.task).then(() => {
+          if (!this.requestInProcess) {
+            this.loading = false;
+            this.task = _.clone(this.cardData);
+          }
+        }); // to update the task state
+      } else {
+        this.loading = true;
+        this.$store.dispatch('todos/changeTaskState', this.task).then(() => {
+          this.$store.dispatch('todos/editTask', this.task).then(() => {
+            if (!this.requestInProcess) {
+              this.loading = false;
+              this.task = _.clone(this.cardData);
+              this.showEditIcon = true;
+            }
+          });
+        });
```

Comment:
Can be drastically simplified.

1. We can use async await to avoid nested .then chaining
2. We can use early returns
```suggestion
      if (this.showEditIcon) {
        this.loading = true;
        await this.$store.dispatch('todos/changeTaskState', this.task)
        
        if (this.requestInProcess) return
        
        this.loading = false;
        this.task = _.clone(this.cardData);
        
        return
      }
      this.loading = true;
      await this.$store.dispatch('todos/changeTaskState', this.task)
      await this.$store.dispatch('todos/editTask', this.task)
      if (this.requestInProcess) return 
              
      this.loading = false;
      this.task = _.clone(this.cardData);
      this.showEditIcon = true;
```

Reply:
Resolved